### PR TITLE
Clang Address Sanitizer and Undefinded Behavior Sanitizer

### LINKED ISSUE #883: Travis/GitLab CI Wishlist
Stuff that could/should be checked in CI
- [ ] Clang sanitizer build (e.g. http://clang.llvm.org/docs/AddressSanitizer.html)
- [ ] Static analysis

### PR DESCRIPTION
**Pull request by mkuron**
Fixes #883

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -374,6 +381,18 @@ endif()
 
 set(CMAKE_MACOSX_RPATH TRUE)
 
+if( WITH_ASAN )
+  set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -g -O1")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fsanitize=address -fno-omit-frame-pointer")
+endif()
+if( WITH_MSAN )
```
Comment (KaiSzuttor) [inline CMakeLists.txt:391]: could you please reflect in the logic that ```WITH_ASAN``` and  ```WITH_MSAN``` are mutually exclusive?

File: src/core/lbgpu.cpp [hunk 2]
```diff
@@ -258,7 +258,7 @@ void lb_reinit_parameters_gpu() {
   for(ii=0;ii<LB_COMPONENTS;++ii){
     lbpar_gpu.mu[ii] = 0.0;
   
-    if (lbpar_gpu.viscosity[ii] > 0.0) {
+    if (lbpar_gpu.viscosity[ii] > 0.0 && lbpar_gpu.agrid > 0.0 && lbpar_gpu.agrid > 0.0 ) {
```
Comment (fweik) [inline src/core/lbgpu.cpp]: Duplicated condition

File: CMakeLists.txt (hunk 1 above)
Comment (mkuron) [reply CMakeLists.txt:391]: Done

**Comment by KaiSzuttor** [comment]
if I compile with the MSAN option, I get
```ImportError: /tikhome/kai/espresso/build/src/core/io/writer/h5md/libH5mdCore.so: undefined symbol: __msan_retval_tls```

**Comment by mkuron** [comment]
> `__msan_retval_tls`

I couldn't find any documentation on how to make MSAN work with shared libraries. It's possible that the Python interpreter needs to be recompiled with MSAN enabled.
It works fine though with the unit tests, so I'll leave the option in for now.

That means we are ready to merge now. Please review, @fweik @KaiSzuttor.

**Comment by fweik** [comment]
LGTM. @KaiSzuttor you can hit the button if you are happy.

**Pull request merged**